hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -161,4 +162,20 @@ protected final void dropIndexOnRestore() {
       }
     }
   }
+
+  protected final void maybeDisableWriteRecordPositions() {
```

Comment:
I don't think we should fix it in the executors, because COW table also share these executors. I kind of think we should fix it in the `HoodieAppendHandle`.

```java
this.shouldWriteRecordPositions = config.shouldWriteRecordPositions()
        // record positions supported only from table version 8
        && config.getWriteVersion().greaterThanOrEquals(HoodieTableVersion.EIGHT);
```

Reply:
Doing it at the `HoodieAppendHandle` level requires instantiating and scanning the timeline at every executor, which is an overhead that should not be introduced I think.

Reply:
can you just do it in the writeClient.upsert ? The entrance of the DAG trigger.

Reply:
can you just do it in the writeClient.upsert ? The entrance of the DAG trigger.

Reply:
Moved the write config modification to the `BaseHoodieWriteClient` and only exercise it in Spark.